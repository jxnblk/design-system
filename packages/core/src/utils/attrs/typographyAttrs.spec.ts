import { typographyAttrs, textStylesValues } from '.'
import { createTheme } from '..'

const theme = createTheme()

describe('typographyAttrs', () => {
  it.each(textStylesValues)('works with and textStyle=%s', (textStyle) => {
    const attrs = typographyAttrs({ textStyle, theme })

    expect(attrs).toMatchSnapshot()
  })

  it('should handle an array prop', () => {
    const attrs = typographyAttrs({ textStyle: ['title3', null, 'title2', null, 'title1'], theme })

    expect(attrs).toMatchSnapshot()
  })
})
