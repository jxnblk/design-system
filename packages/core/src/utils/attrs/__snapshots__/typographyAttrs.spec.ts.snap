// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typographyAttrs should handle an array prop 1`] = `
Object {
  "caps": Array [],
  "fontSize": Array [
    "52px",
    null,
    "64px",
    null,
    "80px",
  ],
  "fontWeight": Array [
    700,
    null,
    700,
    null,
    700,
  ],
  "letterSpacing": Array [
    null,
    null,
  ],
  "lineHeight": Array [
    "64px",
    null,
    "80px",
    null,
    "96px",
  ],
  "textTransform": Array [
    null,
    null,
  ],
}
`;

exports[`typographyAttrs works with and textStyle=article 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 500,
  "letterSpacing": "-0.006rem",
  "lineHeight": "28px",
}
`;

exports[`typographyAttrs works with and textStyle=articleBold 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 700,
  "letterSpacing": "-0.006rem",
  "lineHeight": "28px",
}
`;

exports[`typographyAttrs works with and textStyle=caption 1`] = `
Object {
  "fontSize": "12px",
  "fontWeight": 500,
  "letterSpacing": "0.006rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=captionBold 1`] = `
Object {
  "fontSize": "12px",
  "fontWeight": 700,
  "letterSpacing": "0.006rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=disclaimer 1`] = `
Object {
  "fontSize": "11px",
  "fontWeight": 500,
  "letterSpacing": "0.013rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=disclaimerBold 1`] = `
Object {
  "fontSize": "11px",
  "fontWeight": 700,
  "letterSpacing": "0.013rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=heading1 1`] = `
Object {
  "fontSize": "40px",
  "fontWeight": 700,
  "lineHeight": "48px",
}
`;

exports[`typographyAttrs works with and textStyle=heading2 1`] = `
Object {
  "fontSize": "32px",
  "fontWeight": 700,
  "lineHeight": "40px",
}
`;

exports[`typographyAttrs works with and textStyle=heading3 1`] = `
Object {
  "fontSize": "26px",
  "fontWeight": 700,
  "lineHeight": "32px",
}
`;

exports[`typographyAttrs works with and textStyle=heading4 1`] = `
Object {
  "fontSize": "20px",
  "fontWeight": 700,
  "lineHeight": "24px",
}
`;

exports[`typographyAttrs works with and textStyle=heading5 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 700,
  "lineHeight": "20px",
}
`;

exports[`typographyAttrs works with and textStyle=heading6 1`] = `
Object {
  "fontSize": "14px",
  "fontWeight": 700,
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=highlight 1`] = `
Object {
  "caps": true,
  "fontSize": "10px",
  "fontWeight": 500,
  "letterSpacing": "0.013rem",
  "lineHeight": "12px",
  "textTransform": "uppercase",
}
`;

exports[`typographyAttrs works with and textStyle=highlightBold 1`] = `
Object {
  "caps": true,
  "fontSize": "10px",
  "fontWeight": 700,
  "letterSpacing": "0.013rem",
  "lineHeight": "12px",
  "textTransform": "uppercase",
}
`;

exports[`typographyAttrs works with and textStyle=overline 1`] = `
Object {
  "fontSize": "12px",
  "fontWeight": 700,
  "letterSpacing": "0.063rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=overlineMedium 1`] = `
Object {
  "fontSize": "12px",
  "fontWeight": 500,
  "letterSpacing": "0.063rem",
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=paragraph 1`] = `
Object {
  "fontSize": "14px",
  "fontWeight": 500,
  "letterSpacing": "-0.003rem",
  "lineHeight": "20px",
}
`;

exports[`typographyAttrs works with and textStyle=paragraph2 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 500,
  "letterSpacing": "-0.004rem",
  "lineHeight": "24px",
}
`;

exports[`typographyAttrs works with and textStyle=paragraph2Bold 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 700,
  "letterSpacing": "-0.004rem",
  "lineHeight": "24px",
}
`;

exports[`typographyAttrs works with and textStyle=paragraphBold 1`] = `
Object {
  "fontSize": "14px",
  "fontWeight": 700,
  "letterSpacing": "-0.003rem",
  "lineHeight": "20px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading1 1`] = `
Object {
  "fontSize": "40px",
  "fontWeight": 500,
  "lineHeight": "48px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading2 1`] = `
Object {
  "fontSize": "32px",
  "fontWeight": 500,
  "lineHeight": "40px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading3 1`] = `
Object {
  "fontSize": "26px",
  "fontWeight": 500,
  "lineHeight": "32px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading4 1`] = `
Object {
  "fontSize": "20px",
  "fontWeight": 500,
  "lineHeight": "24px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading5 1`] = `
Object {
  "fontSize": "16px",
  "fontWeight": 500,
  "lineHeight": "20px",
}
`;

exports[`typographyAttrs works with and textStyle=subheading6 1`] = `
Object {
  "fontSize": "14px",
  "fontWeight": 500,
  "lineHeight": "16px",
}
`;

exports[`typographyAttrs works with and textStyle=subtitle1 1`] = `
Object {
  "fontSize": "52px",
  "fontWeight": 500,
  "lineHeight": "64px",
}
`;

exports[`typographyAttrs works with and textStyle=title1 1`] = `
Object {
  "fontSize": "80px",
  "fontWeight": 700,
  "lineHeight": "96px",
}
`;

exports[`typographyAttrs works with and textStyle=title2 1`] = `
Object {
  "fontSize": "64px",
  "fontWeight": 700,
  "lineHeight": "80px",
}
`;

exports[`typographyAttrs works with and textStyle=title3 1`] = `
Object {
  "fontSize": "52px",
  "fontWeight": 700,
  "lineHeight": "64px",
}
`;
